src/sstruct_ls/ssamg_interp.c

```diff
@@ -395,37 +396,56 @@ hypre_SSAMGSetupInterpOp( hypre_SStructMatrix  *A,
                         hypre_IntersectBoxes(tmp_box, hypre_BoxArrayBox(compute_boxes, ii), compute_box);
 
                         hypre_BoxGetStrideSize(compute_box, stride, loop_size);
-                        if (hypre_IndexD(loop_size, cdir) > 1)
+
+                        /* Only do the renormalization at boundaries if interp_type < 0
+                         * (that is, if no unstructured interpolation is used) */
+                        if (interp_type >= 0)
                         {
+                           /* If handling all interpolation at part boundaries with unstructured, zero out the structured weights */
+                           /* WM: todo - this zeros out the boundary weights, but we should not rely on a halo here...
+                            *            Move this later, and loop over the same boxes that get converted for A_u_aug */
                            hypre_BoxLoop1Begin(ndim, loop_size, P_dbox, Pstart, Pstride, Pi);
                            {
-                              HYPRE_Real center;
-
-                              center = Pp1[Pi] + Pp2[Pi];
-                              if (center)
-                              {
-                                 Pp1[Pi] /= center;
-                                 Pp2[Pi] /= center;
-                              }
+                              Pp1[Pi] = 0.0;
```

Comment:
I'm trying to understand the reason for changing the way renormalization is done here

Reply:
Nevermind, understood!